res/controllers/Hercules-DJ-Console-RMX-2-scripts.js

```diff
@@ -1,95 +1,830 @@
+                            engine.setValue('[Channel1]', 'pfl', false);
+                        }
+                    }
+                }
+            };
+        },
+        shift: function() {
+            this.input = function(channel, control, value, status, group) {
+                engine.setValue(this.group, 'eject', value ? 1 : 0);
+                if (value === 0x7F && DJCRMX2.autoPFL) {
+                    engine.setValue(this.group, 'pfl', false);
+                }
+            };
+        },
+    });
+
+    this.ffButton = new components.Button({
+        number: deckNumbers,
+        unshift: function() {
+            this.input = function(channel, control, value, status, group) {
+                engine.setValue(this.group, 'fwd', value);
+            };
+        },
+        shift: function() {
+            this.input = function(channel, control, value, status, group) {
+                if (value === 0x7F) {
+                    engine.brake(this.number, 1);
+                }
+            };
+        },
+    });
+
+    this.rwButton = new components.Button({
+        number: deckNumbers,
+        unshift: function() {
+            this.input = function(channel, control, value, status, group) {
+                engine.setValue(this.group, 'back', value);
+            };
+        },
+        shift: function() {
+            this.input = function(channel, control, value, status, group) {
+                if (value === 0x7F) {
+                    engine.spinback(this.number, 1);
+                }
+            };
+        },
+    });
+
+    this.pitchBendIncrButton = new components.Button({
+        number: deckNumbers,
+        unshift: function() {
+            this.input = function(channel, control, value, status, group) {
+                engine.setValue(this.group, 'rate_temp_up', value);
+            };
+        },
+        shift: function() {
+            this.input = function(channel, control, value, status, group) {
+                var range = engine.getValue(this.group, 'rateRange');
+
+                if ((range * 2) > 0.90) {
+                    range = 0.90;
+                } else {
+                    range = range * 2;
+                }
+
+                if (value === 0x7F) {
+                    engine.setValue(this.group, 'rateRange', range);
+                }
+            };
+        },
+    });
+
+    this.pitchBendDecrButton = new components.Button({
+        number: deckNumbers,
+        unshift: function() {
+            this.input = function(channel, control, value, status, group) {
+                engine.setValue(this.group, 'rate_temp_down', value);
+            };
+        },
+        shift: function() {
+            this.input = function(channel, control, value, status, group) {
+                var range = engine.getValue(this.group, 'rateRange');
+
+                if ((range / 2) < theDeck.setUpSpeedSliderRange) {
+                    range = theDeck.setUpSpeedSliderRange;
```

Comment:
You're right. Your proposals sound better...I will change it to "initialSpeedSliderRange". It was called "setUpSpeed..." because the value came from the Mixxx setup option.